activerecord/lib/active_record/relation.rb

```diff
@@ -773,8 +773,14 @@ def ==(other)
       end
     end
 
-    def pretty_print(q)
-      q.pp(records)
+    def pretty_print(pp)
+      subject = loaded? ? records : annotate("loading for pp")
+      entries = subject.take([limit_value, 11].compact.min)
+
+      ellipsis = Class.new { def pretty_print(q); q.text("..."); end }.new
```

Comment:
We'll be creating this class on every pretty print, we may as well create it once.

Reply:
@gsamokovarov 

The reasoning behind creating it every time is that `pp` will likely be used in dev env, but not in production, hence we don't really need to define it early. I could memoize it though, but I don't think it has a point, should performance matter in `pp`? 

Moreover, I don't know if you had the chance to read it but I've tested memory growth related to creating those classes and I doesn't seem so important. I may have overlooked something there ?

Reply:
On a second read, do we need this whole ceremony to print the ellipsis without quotes? Why not pass the string `"..."` as we do in the `#inspect` method already?

<img width="621" alt="Screenshot 2022-01-07 at 17 12 40" src="https://user-images.githubusercontent.com/604618/148564514-dc4c0bd3-e492-4a63-9894-8c4dfa68e3e5.png">

Reply:
Fair point, it just felt weird seeing that rather than a clean ellipsis, but whatever!

Reply:
Thank you, looks good!